### DISCUSSION TITLE
FMData - authentication error 503 on FMS 19 & 20

### DISCUSSION THREAD

**Discussion by isquareIT** [fuzzzerd/fmdata#348] [Q&A] [answered]
Hi, a simple Find request on the exact, same database throws an exception "Cannot authenticate" on FMS 19 and 20 while on FMS 18 works fine. Under the debugger, the response code is 503 (200 on FMS 18) at line 214 in function "RefreshTokenAsync" of file "FileMakerRestClient.cs".
Could you please advise.

**Comment by fuzzzerd** [comment, marked as answer]
Are you able to successfully perform Data API queries using Postman or Curl? HTTP 503 is a Service Unavailable which is not related to FMData, that is a server configuration issue. 

The [RefreshTokenAsync checks for 200 and 401 responses](https://github.com/fuzzzerd/fmdata/blob/2e3c22cf5fac9e65156b435531e2f4042ba1992a/src/FMData.Rest/FileMakerRestClient.cs#L200) and if it doesn't get one of those it throws a "Could not authenticate" exception.

**Reply by isquareIT** [reply]
Yes, you are totally right. The 503 (Service Unavailable) was due in fact to the service being unavailable!
I switched on the Data API service on the FMservers 19 and 20 and everything works fine now.
Thank you so much for the swift reply.

**Reply by fuzzzerd** [reply]
Glad its working as expected!

**Comment by isquareIT** [comment]
There was no issue really. FMS' Data API was left switched off. No problem with the code.
